server/game/cards/characters/01/oldforesthunter.js

```diff
@@ -4,30 +4,16 @@ class OldForestHunter extends DrawCard {
     setupCardAbilities(ability) {
         this.action({
             title: 'Discard a card to gain 1 gold',
-            method: 'discard',
-            limit: ability.limit.perPhase(1)
+            condition: () => !this.controller.cannotGainGold,
+            cost: ability.costs.discardFromHand(),
+            limit: ability.limit.perPhase(1),
+            handler: context => {
+                this.game.addGold(this.controller, 1);
```

Comment:
Would work in this case since things are synchronous (at least for now). Doesn't work in the general case (async things like discarding cards, killing characters, leaving play, etc).

In any case, adding the cannot-gain-gold check to the triggering condition for appropriate cards sidesteps this problem, acting as an upfront if for the ability.